common/utils.js

```
@@ -172,18 +172,42 @@
         notNull: "<i>All Records With Value</i>"
     })
 
-    .factory('UriUtils', ['$injector', '$rootScope', '$window', 'appContextMapping', 'appTagMapping', 'ContextUtils', 'Errors', 'messageMap', 'parsedFilter',
-        function($injector, $rootScope, $window, appContextMapping, appTagMapping, ContextUtils, Errors, messageMap, ParsedFilter) {
-        var chaiseConfig = Object.assign({}, $rootScope.chaiseConfig);
+    .factory('UriUtils', ['appContextMapping', 'appTagMapping', 'ConfigUtils', 'ContextUtils', 'defaultChaiseConfig', 'Errors', 'messageMap', 'parsedFilter', '$injector', '$rootScope', '$window',
+        function(appContextMapping, appTagMapping, ConfigUtils, ContextUtils, defaultChaiseConfig, Errors, messageMap, ParsedFilter, $injector, $rootScope, $window) {
+        var chaiseBaseURL;
+
+        function getCatalogId() {
+            var chaiseConfig = ConfigUtils.getConfigJSON();
+
+            if ($rootScope.context && $rootScope.context.catalogID) {
+                return "" + $rootScope.context.catalogID;
+            }
+
+            // NOTE: what if it's '?' instead of '#'
+            if ($window.location.hash != "") {
+                return getCatalogIDFromLocation();
+            }
+
+            if (chaiseConfig.defaultCatalog) {
+                return "" + chaiseConfig.defaultCatalog;
+            }
+
+            return "";
+        }
 
         /**
          * Returns the catalog id
          * TODO we might want to refactor the caller of this function
          * @return {String}
          */
         function getCatalogIDFromLocation() {
-            var hash = getLocationHash($window.location);
-            return hash.split('/')[0].slice(1);
+            try {
+                var hash = getLocationHash($window.location);
+                return hash.split('/')[0].slice(1);
+            } catch (exception) {
+                // pass to error handler
+                throw exception;
```

This should fail silently as I explained in the other comment.